apps/snitch_api/lib/snitch_api_web/controllers/taxon_controller.ex

```diff
@@ -0,0 +1,32 @@
+defmodule SnitchApiWeb.TaxonController do
+  use SnitchApiWeb, :controller
+
+  alias Snitch.Repo
+  alias Snitch.Data.Schema.Taxon
+
+  def index(conn, _params) do
+    taxons =
+      Repo.all(Taxon)
```

Comment:
Why using `Repo` here.

```elixir
alias Snitch.Data.Model.TaxonModel
TaxonModel.get_all |> Repo.preload([:parent, :taxonomy])
```

Reply:
I saw that. Initially, I tried like you suggested. But, I did not find the model in the `develop` branch for `snitch_core`. But, it is present in the `super_sonic` branch. Need to merge those files. IMO :idea:
https://github.com/aviabird/snitch/tree/develop/apps/snitch_core/lib/core/data/model

Reply:
Copy that model code as it is in your branch. Use `super_sonic` as reference. And see if you need to copy any code from there.